manifests/init.pp

```diff
@@ -0,0 +1,48 @@
+# @summary Profile for installing and managing Java.
+#
+# Profile for installing and managing Java.
+#
+# @param install_lock_files
+#   Boolean - Install lock files or not.
+#
+# @param lock_files
+#   Hash of lock files (and parent dir(s)).
+#
+# @param pkg_list
+#   Hash of package names to install
+#   Keys must be an OS Family name
+#   Values must be a Hash of the format:
+#       Key = package name
+#           A package name can be prefixed with "-" to remove it from the list.
+#           In this way, a higher level hiera layer can remove a package added
+#           by a lower, common layer.
+#
+#       Value (optional):
+#           1. <EMPTY> in which case, the package will be installed
+#           2. <HASH> conataining valid attributes for the 
+#              "package" native puppet resource type
+#
+# @example
+#   include profile_java
+class profile_java (
+  Boolean $install_lock_files,
+  Hash    $lock_files,
```

Comment:
I don't see the point of this parameter continuing to be a Hash. I think it can be a simple array.

Reply:
I think you're commenting on $lock_files here (but maybe this was just another comment about $pkg_list?):

Since I'm feeding $lock_files to ensure_resources, this has to be a hash (2nd arg to ensure_resources has to a hash). Plus, we need some of these items to be directories, not files, and so I _think_ we need to pass include that at a minimum in those cases in addition to the path.

Reply:
Oops. I intended to be commenting on the `pkg_list` parameter here.